internal/pkg/skuba/addons/cilium.go

```diff
@@ -87,235 +87,524 @@ func (ciliumCallbacks) afterApply(addonConfiguration AddonConfiguration, skubaCo
-          [{"key":"dedicated","operator":"Equal","value":"master","effect":"NoSchedule"}]
+        scheduler.alpha.kubernetes.io/critical-pod: ""
+        scheduler.alpha.kubernetes.io/tolerations: '[{"key":"dedicated","operator":"Equal","value":"master","effect":"NoSchedule"}]'
+      labels:
+        k8s-app: cilium
+        kubernetes.io/cluster-service: "true"
     spec:
-      serviceAccountName: cilium
-      initContainers:
-      - name: install-cni-conf
-        image: {{.CiliumImage}}
-        command:
-          - /bin/sh
-          - "-c"
-          - "cp -f /etc/cni/net.d/10-cilium-cni.conf /host/etc/cni/net.d/10-cilium-cni.conf"
-        volumeMounts:
-        - name: host-cni-conf
-          mountPath: /host/etc/cni/net.d
-      - name: install-cni-bin
-        image: {{.CiliumImage}}
-        command:
-          - /bin/sh
-          - "-c"
-          - "cp -f /usr/lib/cni/* /host/opt/cni/bin/"
-        volumeMounts:
-        - name: host-cni-bin
-          mountPath: /host/opt/cni/bin/
-      - name: clean-cilium-state
-        image: {{.CiliumInitImage}}
-        imagePullPolicy: IfNotPresent
+      containers:
+      - args:
+        - --config-dir=/tmp/cilium/config-map
         command:
-        - cilium-init
+        - cilium-agent
         env:
-        - name: CLEAN_CILIUM_STATE
+        - name: K8S_NODE_NAME
+          valueFrom:
+            fieldRef:
+              apiVersion: v1
+              fieldPath: spec.nodeName
+        - name: CILIUM_K8S_NAMESPACE
+          valueFrom:
+            fieldRef:
+              apiVersion: v1
+              fieldPath: metadata.namespace
+        - name: CILIUM_FLANNEL_MASTER_DEVICE
           valueFrom:
             configMapKeyRef:
-              key: clean-cilium-state
+              key: flannel-master-device
               name: cilium-config
               optional: true
-        - name: CLEAN_CILIUM_BPF_STATE
+        - name: CILIUM_FLANNEL_UNINSTALL_ON_EXIT
           valueFrom:
             configMapKeyRef:
-              key: clean-cilium-bpf-state
+              key: flannel-uninstall-on-exit
               name: cilium-config
               optional: true
-        securityContext:
-          capabilities:
-            add:
-            - NET_ADMIN
-          privileged: true
-        volumeMounts:
-        - mountPath: /var/run/cilium
-          name: cilium-run
-      containers:
-      - image: {{.CiliumImage}}
+        - name: CILIUM_CLUSTERMESH_CONFIG
+          value: /var/lib/cilium/clustermesh/
+        - name: CILIUM_CNI_CHAINING_MODE
+          valueFrom:
+            configMapKeyRef:
+              key: cni-chaining-mode
+              name: cilium-config
+              optional: true
+        - name: CILIUM_CUSTOM_CNI_CONF
+          valueFrom:
+            configMapKeyRef:
+              key: custom-cni-conf
+              name: cilium-config
+              optional: true
+        image: "docker.io/cilium/cilium:v1.6.6"
```

Comment:
Can you please adjust to `registry.suse.com`

Reply:
Good catch, sorry for that :man_facepalming: